ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1026,6 +1026,17 @@ func (r *RayClusterReconciler) reconcileAutoscalerServiceAccount(instance *rayio
 			return err
 		}
 
+		// If users specify ServiceAccountName for the head Pod, they need to create a ServiceAccount themselves.
+		// However, if KubeRay creates a ServiceAccount for users, the autoscaler may encounter permission issues during
+		// zero-downtime rolling updates when RayService is performed. See https://github.com/ray-project/kuberay/issues/1123
+		// for more details.
+		if instance.Spec.HeadGroupSpec.Template.Spec.ServiceAccountName == namespacedName.Name {
+			r.Log.Error(err, fmt.Sprintf(
+				"If users specify ServiceAccountName for the head Pod, they need to create a ServiceAccount themselves."+
+					"However, ServiceAccount %s is not found. Please create one.", namespacedName.Name), "ServiceAccount", namespacedName)
```

Comment:
@kevin85421 just curious, where are we checking whether the serviceaccount exists?

Reply:
Nit, we should make the call-to-action ("Please create one") in the error message as foolproof as possible so that users with varying levels of expertise can unblock themselves.  Is there a link to a doc we can include, or a very brief example of how to create a service account?

Reply:
Updated https://github.com/ray-project/kuberay/pull/1128/commits/5a9e3e28ac3f9123a1a52a731057216f79e77c6a.